src/contexts/InputContext.js

```diff
@@ -0,0 +1,43 @@
+import { createContext, useState, useContext } from "react";
+import { DataContext } from "./DataContext";
+
+export const InputContext = createContext();
+
+const InputContextProvider = (props) => {
+
+  const { inputWeatherValue } = useContext(DataContext);
+
+  const [weekday, setWeekday] = useState('');
+  const [precip, setPrecip] = useState(null);
+  const [wind, setWind] = useState(null);
+
+  const weatherSubmit = (e) => {
+    e.preventDefault();
+    inputWeatherValue(weekday, precip, wind);
+    setWeekday('');
+    setPrecip(null);
+    setWind(null);
+  }
+
+  const takeWeekdayInput = (e) => setWeekday(e.target.value);
```

Comment:
this can be 
`const takeWeekdayInput = ({ target: { value }}) => setWeekday(value)`